packages/d3fc-discontinuous-scale/src/tickFilter.js

```diff
@@ -1,11 +1,19 @@
+import { pairs } from 'd3-array';
+
+function areEqual([a, b]) {
+    if (a.valueOf) {
+        return a.valueOf() == b.valueOf();
+    } else {
+        return a == b;
+    }
+}
+
 export default function tickFilter(ticks, discontinuityProvider) {
-    const discontinuousTicks = [];
-    for (const tick of ticks) {
-        const up = discontinuityProvider.clampUp(tick);
-        const down = discontinuityProvider.clampDown(tick);
-        if (up === down) {
-            discontinuousTicks.push(up);
-        }
+    const discontinuousTicks = ticks.map(discontinuityProvider.clampUp);
+    if (pairs(discontinuousTicks).some(areEqual)) {
```

Comment:
How about `discontinuousTicks.length !== new Set(discontinousTicks.map(d => d?.valueOf())).length`?

(You can drop the new dependency in that case)